src/migrations/20171129064057_multicontainer.js

```diff
@@ -0,0 +1,311 @@
+
+var defaultLegacyVolume = function (appId) {
+	return `resin-app-${appId}`
+}
+
+var tryParse = function (obj) {
+	try {
+		return JSON.parse(obj)
+	}
+	catch(e) {
+		return {}
+	}
+}
+
+var singleToMulticontainerApp = function (app) {
+	// From *very* old supervisors, env or config may be null
+	// so we ignore errors parsing them
+	let conf = tryParse(app.config)
+	let env = tryParse(app.env)
+	let environment = {}
+	let appId = parseInt(app.appId)
+	for (let key in env) {
+		if (!/^RESIN_/.test(key)) {
+			environment[key] = env[key]
+		}
+	}
+	let newApp = {
+		appId: appId,
+		commit: app.commit,
+		name: app.name,
+		releaseId: 1,
+		networks: {},
+		volumes: {}
+	}
+	let defaultVolume = defaultLegacyVolume(appId)
+	newApp.volumes[defaultVolume] = {}
+	let updateStrategy = conf['RESIN_SUPERVISOR_UPDATE_STRATEGY']
+	if (updateStrategy == null) {
+		updateStrategy = 'download-then-kill'
+	}
+	let handoverTimeout = conf['RESIN_SUPERVISOR_HANDOVER_TIMEOUT']
+	if (handoverTimeout == null) {
+		handoverTimeout = ''
+	}
+	let restartPolicy = conf['RESIN_APP_RESTART_POLICY']
+	if (restartPolicy == null) {
+		restartPolicy = 'always'
+	}
+	newApp.services = [
```

Comment:
@dfunckt could you check this composition out, you seemed good at spotting issues with them in the api :P